### PR TITLE
Enable read-only filesystem when assets do not need to be downloaded

### LINKED ISSUE #126: Enable assets manager to run on read-only filesystem when using LocalStorageDriver
# Current state

During asset creation development process, or if you want to embed assets in a container at build time, 
Modelkit can be configured to use StorageProvider with LocalStorageDriver and a AssetsDir pointing to the same location of the LocalStorage.
This allow to only use local file and do not duplicate them on local file system.

However, modelkit still requires to have write permission on the filesystem because the management of lock files is not disabled in this condition.

# Fix

Idea is to disable the lock files management in this configuration.
It is fine to do that since we are guaranteed that we will not download anything in this configuration

### PR DESCRIPTION
**Pull request by ldeflandre**
Closes #126

### CODE REVIEW

**Comment by CyrilLeMat** [comment]
LGTM but I would like someone else to review (@victorbenichoux , @tgenin  or @antoinejeannot )

File: modelkit/assets/manager.py [hunk 1]
```diff
@@ -81,74 +81,79 @@ def __init__(
...
-                with ContextualizedLogging(version=spec.version):
-                    return self._fetch_asset_version(
-                        spec, local_versions, _force_download
-                    )
 
-            remote_versions = []
-            if self.storage_provider:
-                remote_versions = self.storage_provider.get_versions_info(spec.name)
-                logger.debug("Fetched remote versions", remote_versions=remote_versions)
+            self._resolve_version(spec)
```
Comment (ldeflandre) [inline modelkit/assets/manager.py:85]: logic of resolution and download of assets were mixed and benefits from being split formally here.

File: tests/assets/test_local_manager.py [hunk 2]
```diff
@@ -244,42 +238,34 @@ def test_read_only_manager_with_fetch(
 
 @pytest.mark.parametrize(*test_versioning.INIT_VERSIONING_PARAMETRIZE)
 def test_fetch_asset_version_no_storage_provider(
-    version_asset_name, version, versioning, working_dir
+    version_asset_name, version, versioning
 ):
-    manager = AssetsManager(assets_dir=working_dir)
+    manager = AssetsManager(
+        assets_dir=os.path.join(TEST_DIR, "testdata", "test-bucket", "assets-prefix")
```
Comment (ldeflandre) [inline tests/assets/test_local_manager.py:244]: with no storage provider, assets dir need to be pre-filled

File: docs/assets/storage_provider.md [hunk 3]
```diff
@@ -80,13 +80,22 @@ The client is created by passing the authentication information to `BlobServiceC
...
+If you would like to run on already downloaded assets.
 
 | Environment variable             | Notes                 |
 | -------------------------------- | --------------------- |
 | `MODELKIT_STORAGE_BUCKET`        | path to the local folder |
+| `MODELKIT_STORAGE_PREFIX`        | sub directory where assets are fetched from |
+
+### Pre-downloaded mode
+
+Use `MODELKIT_STORAGE_PROVIDER=` to use only assets available in assets dir.
```
Comment (tgenin) [inline docs/assets/storage_provider.md:96]: or `unset  MODELKIT_STORAGE_PROVIDER` ?

Comment (ldeflandre) [reply docs/assets/storage_provider.md:96]: Yes, it works as well.

**Comment by tgenin** [comment]
Looks good to me

**Review by tgenin — APPROVED** [review]

**Pull request merged**